src/mesh/coordinates.cxx

```diff
@@ -210,17 +210,27 @@ Coordinates::Coordinates(Mesh* mesh, Options* options)
   // 'interpolateAndExtrapolate' to set them. Ensures that derivatives are
   // smooth at all the boundaries.
 
-  const bool extrapolate_x = (*options)["extrapolate_x"].withDefault(not mesh->sourceHasXBoundaryGuards());
-  const bool extrapolate_y = (*options)["extrapolate_y"].withDefault(not mesh->sourceHasYBoundaryGuards());
+  bool extrapolate_x = not mesh->sourceHasXBoundaryGuards();
+  bool extrapolate_y = not mesh->sourceHasYBoundaryGuards();
+
+  // Allow extrapolation to be replaced by nearest-grid point value according to option
+  if (extrapolate_x) {
+    extrapolate_x = (*options)["extrapolate_x"].withDefault(true);
+  }
+  if (extrapolate_y) {
+    extrapolate_y = (*options)["extrapolate_y"].withDefault(true);
+  }
```

Comment:
I don't see the benefit of this rather than the version I wrote. I am biased though

Reply:
Is this to prevent someone from extrapolating if the guard cells are present?

Reply:
Sorry, I was not thinking straight when I made this. I was thinking the nearest-grid-cell thing would override the grid file values if `extrapolate_y=false` was set explicitly, but that isn't true - when `extrapolate_y=false` the grid file values are used if present, and nearest-grid-cell value if they are not. So there is no benefit to this PR. Closing now...